lib/semantics/resolve-labels.cc

```diff
@@ -0,0 +1,969 @@
+  void CheckName(const parser::BlockConstruct &blockConstruct) {
+    if (!BothEqOrNone(
+            std::get<parser::Statement<parser::BlockStmt>>(blockConstruct.t)
+                .statement.v,
+            std::get<parser::Statement<parser::EndBlockStmt>>(blockConstruct.t)
+                .statement.v)) {
+      Report(eh, currentPosition_,
+          parser::MessageFormattedText{
+              "BLOCK construct name mismatch"_err_en_US});
+    }
+  }
+  /// \brief Check <i>team-cosntruct-name</i>, constraint C1112
+  void CheckName(const parser::ChangeTeamConstruct &changeTeamConstruct) {
+    if (!BothEqOrNone(std::get<std::optional<parser::Name>>(
+                          std::get<parser::Statement<parser::ChangeTeamStmt>>(
+                              changeTeamConstruct.t)
+                              .statement.t),
+            std::get<std::optional<parser::Name>>(
+                std::get<parser::Statement<parser::EndChangeTeamStmt>>(
+                    changeTeamConstruct.t)
+                    .statement.t))) {
+      Report(eh, currentPosition_,
+          parser::MessageFormattedText{
+              "CHANGE TEAM construct name mismatch"_err_en_US});
+    }
+  }
+
+  // -----------------------------------------------
+  // Case 2: same as case 1, but subblock statement construct-names are
+  // optional but if they are specified their values must be identical
+
+  /// \brief Check <i>if-construct-name</i>
+  /// Constraint C1142 - opening and ending name must match if present, and
+  /// <i>else-if-stmt</i> and <i>else-stmt</i> must either match or be unnamed
+  void CheckName(const parser::IfConstruct &ifConstruct) {
+    const auto &constructName{std::get<std::optional<parser::Name>>(
+        std::get<parser::Statement<parser::IfThenStmt>>(ifConstruct.t)
+            .statement.t)};
+    if (!BothEqOrNone(constructName,
+            std::get<parser::Statement<parser::EndIfStmt>>(ifConstruct.t)
+                .statement.v)) {
+      Report(eh, currentPosition_,
+          parser::MessageFormattedText{"IF construct name mismatch"_err_en_US});
+    }
+    for (const auto &elseIfBlock :
+        std::get<std::list<parser::IfConstruct::ElseIfBlock>>(ifConstruct.t)) {
+      if (!PresentAndEq(
+              std::get<std::optional<parser::Name>>(
+                  std::get<parser::Statement<parser::ElseIfStmt>>(elseIfBlock.t)
+                      .statement.t),
+              constructName)) {
+        Report(eh, currentPosition_,
+            parser::MessageFormattedText{
+                "ELSE IF statement name mismatch"_err_en_US});
+      }
+    }
+    if (std::get<std::optional<parser::IfConstruct::ElseBlock>>(ifConstruct.t)
+            .has_value()) {
+      if (!PresentAndEq(
+              std::get<parser::Statement<parser::ElseStmt>>(
+                  std::get<std::optional<parser::IfConstruct::ElseBlock>>(
+                      ifConstruct.t)
+                      .value()
```

Comment:
If you're grepping and replacing `.value().` with `->` as suggested above, don't miss this site with the extra newline.

Reply:
syntax changes made